remote-query/remote-query-server/src/test/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptorTest.java

```diff
@@ -268,6 +272,38 @@ public void testStatusAfterRemove() {
       assertNoTransactionsAndLocks();
    }
 
+   public void testUnsupportedCommands() {
+      Cache<String, String> cache0 = cache(0);
+
+      assertTrue(cache0.isEmpty());
+
+      try {
+         cache0.compute("test.proto", (k, v) -> "import \"missing.proto\";");
+      } catch (Exception e) {
+         assertTrue(e instanceof CacheException);
+         assertTrue(e.getMessage().contains("ISPN028014"));
+      }
+
+      assertTrue(cache0.isEmpty());
+
+      try {
+         FunctionalMap.ReadWriteMap<String, String> rwMap = ReadWriteMapImpl.create(FunctionalMapImpl.create(cache0.getAdvancedCache()));
+
+         rwMap.eval("test.proto", "val", (value, view) -> {
+            view.set(value);
+            return "ret";
+         }).join();
+
+      } catch (CompletionException e) {
+         assertTrue(e.getCause() instanceof CacheException);
+         assertTrue(e.getCause().getMessage().contains("ISPN028014"));
```

Comment:
You could use `Exceptions.expectException(CacheException.class, rwMap.eval(...))`, although there's no overload with a `messageRegex` parameter yet it should be easy to add.

But I also wonder about the error code, I found https://github.com/infinispan/infinispan/blob/81b60c6f22dbfe035a4c004a678606d60544cd00/counter/src/main/java/org/infinispan/counter/logging/Log.java#L65

Reply:
That logging id belongs to query. Counter module came 6 months later and used the same id range, not sure how that happened, and worse, some of those ids are used in commons too.